### DISCUSSION TITLE
.net 7.0 library to get file MIMETYPE

### DISCUSSION THREAD

**Discussion by shiweizhangMSFT** [dotnet/aspnetcore#48906] [Q&A]
Hello, I am looking for a .net 7.0 library to get file MIMETYPE. It should get the file mime type from file binaries, not file extension which is not reliable. By searching on website, I got to know urlmon.dll (https://stackoverflow.com/questions/1654846/in-c-how-can-i-know-the-file-type-from-a-byte), which however looks outdated. I would also prefer to use .net organic libraries to achieve this. If no such library exists, any idea on how to build this functionality is also welcomed. Thanks a lot in advance!

**Comment by gfoidl** [comment]
https://github.com/matheusz2/Mime-Detective-MO looks promising (I didn't try it out).

---

Please note that this repository for development for ASP.NET Core, so for questions like this you may get more specific help on https://stackoverflow.com/ and similar.
